src/common/utxobased/keymanager/keymanager.ts

```diff
@@ -387,33 +387,32 @@ function bip32NetworkFromCoin(
 }
 
 export function verifyAddress(args: VerifyAddressArgs): VerifyAddressEnum {
-  const coinClass = getCoinFromString(args.coin)
-  const coinPrefixes = coinClass.mainnetConstants
-  const maxLength = getMaxLengthOfCoinPrefixes(coinPrefixes)
-  for (let prefixIndex = 0; prefixIndex < maxLength; prefixIndex++) {
-    try {
-      const network: BitcoinJSNetwork = bip32NetworkFromCoin({
-        coinString: args.coin,
-        prefixIndex
-      })
-      guessAddressTypeFromAddress(args.address, network, args.coin, undefined)
-      if (prefixIndex === 0) {
-        return VerifyAddressEnum.good
-      }
-      // All indices above 0 are treated as legacy
-      return VerifyAddressEnum.legacy
-    } catch (e) {
-      if (e instanceof Error) {
+  const coin = getCoinFromString(args.coin)
```

Comment:
This is not what I was thinking about, lets discuss it tomorrow.

Reply:
You're suggestion is awesome! Implement like this:

```
const xprivToXPubRaw = (args: XPrivToXPubArgs, prefixIndex: number): string => {
  const network: BitcoinJSNetwork = bip32NetworkFromCoin({
    coinString: args.coin,
    sigType: args.type,
    prefixIndex,
  })
  const coin = getCoinFromString(args.coin)
  const bip32FromBase58Func = coin.bip32FromBase58Func ?? bip32.fromBase58
  return bip32FromBase58Func(args.xpriv, network).neutered().toBase58()
}

export const xprivToXPub_ = (...arg):string => filterCoinPrefixes(xprivToXPubRaw, 'Invalid private key')(...arg)
```